File: PrefixSum/2845. Count of Interesting Subarrays.py
class Solution:
    def countInterestingSubarrays(self, nums: List[int], modulo: int, k: int) -> int:
        l = [int(x%modulo==k) for x in nums]
        ret = 0
        cur = 0
        d = collections.defaultdict(int)
        d[0] = 1
        for x in l:
            cur += x
            cur %= modulo
            ret += d[(cur - k) % modulo]
            d[cur] =  d[cur] + 1
        return ret

File: PrefixSum/2845. Count of Interesting Subarrays.cpp
class Solution {
public:
    long long countInterestingSubarrays(vector<int>& nums, int modulo, int k) {
        int n = nums.size();
        vector<int>dp(n, 0);
        for (int i = 0; i < n ; i++) {
            if (nums[i] % modulo == k) dp[i] = 1 ,cout << '1' << '\n';
        }
        map<int, int>mp;
        mp[0]++;
        long long ret = 0, cur = 0;
        for (auto x : dp) {
            cur += x;
            cur %= modulo;
            ret += mp[(cur - k + modulo) % modulo];
            mp[cur]++;
        }
        // cout << cur << '\n';
        return ret;
    }
};